src/Microsoft.VisualStudio.Services.Agent/JobServerQueue.cs

```diff
@@ -532,26 +529,25 @@ private async Task ProcessTimelinesUpdateQueueAsync(bool runOnce = false)
                     }
                     else
                     {
-                        if (mainTimelineRecordsUpdateErrors.Count > 0 &&
+                        if (ForceDrainTimelineQueue)
+                        {
+                            ForceDrainTimelineQueue = false;
+                        }
+                    }
+                }
+
+                if (runOnce)
+                {
+                    if (mainTimelineRecordsUpdateErrors.Count > 0 &&
                             _bufferedRetryRecords.ContainsKey(_jobTimelineId) &&
                             _bufferedRetryRecords[_jobTimelineId] != null &&
                             _bufferedRetryRecords[_jobTimelineId].Any(r => r.Variables.Count > 0))
-                        {
-                            Trace.Info("Fail to update timeline records with output variables. Throw exception to fail the job since output variables are critical to downstream jobs.");
-                            throw new AggregateException(StringUtil.Loc("OutputVariablePublishFailed"), mainTimelineRecordsUpdateErrors);
-                        }
-                        else
-                        {
-                            if (ForceDrainTimelineQueue)
-                            {
-                                ForceDrainTimelineQueue = false;
-                            }
-                            if (runOnce)
-                            {
-                                break;
-                            }
-                        }
+                    {
+                        Trace.Info("Fail to update timeline records with output variables. Throw exception to fail the job since output variables are critical to downstream jobs.");
```

Comment:
Can we make sure that this line is always printed (preferably as an error) in case of timeline record update failures?
In which file of agent logs will this line be printed at in case it happens?

Reply:
This line is always printed in the worker log file if an exception occurs at the last stage of the job.